public/app/features/dashboard-scene/settings/variables/VariableEditorForm.tsx

```diff
@@ -19,15 +19,25 @@ import { VariableNameConstraints } from 'app/features/variables/editor/types';
 import { VariableTypeSelect } from './components/VariableTypeSelect';
 import { EditableVariableType, getVariableEditor, hasVariableOptions, isEditableVariableType } from './utils';
 
+const RESERVED_GLOBAL_VARIABLE_NAME_REGEX = /^(?!__).*$/;
+const WORD_CHARACTERS_REGEX = /^\w+$/;
+
 interface VariableEditorFormProps {
   variable: SceneVariable;
+  allVariables: SceneVariable[];
   onTypeChange: (type: EditableVariableType) => void;
   onGoBack: () => void;
   onDelete: (variableName: string) => void;
 }
-
-export function VariableEditorForm({ variable, onTypeChange, onGoBack, onDelete }: VariableEditorFormProps) {
+export function VariableEditorForm({
+  variable,
+  allVariables,
```

Comment:
Rather than passing all variables here, can we pass validation function that would result in `[boolean, errorText]` for instance?

I find it awkward that individual variable form needs to depend on all other existing variables.

Reply:
@dprokop Adjusted by passing `isVariableNameTaken` instead of `allVariables`. `isVariableNameTaken` is also much more stable across re-renders (never changes with useCallback).